src/main/java/chess/domain/chessboard/Chessboard.java

```diff
@@ -108,11 +124,96 @@ private void movePieceToTarget(final Square source, final Square target) {
         chessboard.remove(source);
     }
 
+    public boolean catchKing() {
+        long count = chessboard.values().stream()
+                .filter(piece -> piece.isTypeOf(PieceType.KING))
+                .count();
+        return count != INITIAL_KING_COUNT;
+    }
+
+    public double totalScoreOf(Color color) {
+        Set<Piece> samColorPieces = findSameAllyPieces(color);
+        double exceptPawnScore = calculateTotalScoreExceptPawn(samColorPieces);
+        Set<Piece> pawns = filterPawns(samColorPieces);
+        return pawns.stream()
+                .reduce(exceptPawnScore,
+                        (res, pawn) -> calculatePawnScore(res, pawn, pawns),
+                        (prev, next) -> next);
```

Comment:
1. 스트림의 최종연산에는 sum()이라는 연산자도 있으니 활용해보셔요. (map을 통해 pieces들의 점수들로 변환해두고 최종적으로 sum하기)

reduce는 점진적으로 쌓아나가는 연산인데, 이게 증감일지 감소일지 곱일지는 짜기 나름입니다. 그런데 calculatePawnScore 내부에 증가가 감춰져있으니 첫눈에 봤을 때 이게 뭐하는 로직인지 모르겠어요.
기본값 세팅을 위해 reduce를 그래도 활용하시려면 아래처럼도 가능할 것 같습니다.
```suggestion
        return pawns.stream()
                .map(it -> calculatePawnScore(it, pawns))
                .reduce(exceptPawnScore,
                        Double::sum);
    }

    private Double calculatePawnScore(Piece pawn, Set<Piece> pawns) {
        if (hasSameFileIn(pawns, pawn.getLocatedFile())) {
            return 0.5;
        }
        return 1.0;
    }
```

2. 기존에 작성해주신 코드 중 세번째 인자는 병렬 처리 때 combine하기 위한 함수인데, 병렬 스트림을 사용하지 않으셨으니 굳이 작성할 필요가 없습니다.